src/Compilers/Test/Utilities/CSharp/CSharpTestSource.cs

```diff
@@ -83,5 +84,17 @@ public SyntaxTree[] GetSyntaxTrees(CSharpParseOptions parseOptions, string sourc
         public static implicit operator CSharpTestSource(List<SyntaxTree> source) => new CSharpTestSource(source.ToArray());
         public static implicit operator CSharpTestSource(ImmutableArray<SyntaxTree> source) => new CSharpTestSource(source.ToArray());
         public static implicit operator CSharpTestSource(CSharpTestSource[] source) => new CSharpTestSource(source);
+
+        // Dummy IEnumerable support to satisfy the collection expression and CollectionBuilder requirements
+        IEnumerator<CSharpTestSource> IEnumerable<CSharpTestSource>.GetEnumerator() => throw new NotImplementedException();
+        IEnumerator IEnumerable.GetEnumerator() => throw new NotImplementedException();
+
+        internal static class CSharpTestSourceBuilder
```

Comment:
Why does this need to be a separate type? Why not just make it a static member on this type?

Reply:
It does not *need* to be a separate type, but it is customary for the collection and the collection builder to be different types, and I think it makes sense. See all the collection tests

Reply:
> but it is customary for the collection and the collection builder to be different types

This is a brand new pattern, there's no "customary" at all 😆. I'd just as soon keep it simpler.

Reply:
All our collection builder tests beg to differ. The spec also calls out [two different concepts](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#create-methods): a "collection type" and a "builder type". It is clearer not to blend them and doesn't seem "simpler".
In any case, this seems a nitpick/preference, not a reason to hold this change up. Thanks